### PR TITLE
fix(appconnect): Backfill app_id_str

### PR DESCRIPTION
**Pull request by loewenheim**
This is step 2 in the effort to change the type of `AppConnectBuild.app_id` from an integer to a string.

It backfills the new `app_id_str` column from the existing `app_id` field.

Depends on https://github.com/getsentry/sentry/pull/40261.

### CODE REVIEW

File: src/sentry/migrations/0326_backfill_app_id_str.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_app_id_str(apps, schema_editor):
+    AppConnectBuild = apps.get_model("sentry", "AppConnectBuild")
+
+    for appconnect_build in RangeQuerySetWrapperWithProgressBar(AppConnectBuild.objects.all()):
+        appconnect_build.app_id_str = str(appconnect_build.app_id)
+        appconnect_build.save()
```
Comment (wedamija) [inline src/sentry/migrations/0326_backfill_app_id_str.py:14]: Nit: You can use `.save(update_fields=["app_id_str"])` to just update the specific field.

**Review by wedamija — APPROVED** [review]
This looks safe to me. This table has 188k rows, should be fine to run this live rather than asking ops to do so.

File: src/sentry/migrations/0326_backfill_app_id_str.py (hunk 1 above)
Comment (loewenheim) [reply src/sentry/migrations/0326_backfill_app_id_str.py:14]: Does that do the int/str conversion automatically?

**Comment by wedamija** [comment]
Looks good, just want to note that you should be careful to deploy each of these prs separately

**Pull request merged**